### DISCUSSION TITLE
Test for generic type definition

### DISCUSSION THREAD

**Discussion by HeSitates** [nunit/nunit#4169] [Q&A]
Is there a better way to test if type is instance of a generic type then this?
`Assert.That(sut.GetType().GetGenericTypeDefinition() == typeof(Result<>), Is.True);`

If Result is defined as:
```
public class Result<T>
{
  public bool Success { get; set; }

  public T Value { get; set; }

  public string ErrorMessage { get; set; }
}
```

**Comment by mikkelbu** [comment]
You can write the assert a little more succinct as `Assert.IsInstanceOf(typeof(Result<>), sut);`

**Reply by HeSitates** [reply]
Sadly that does not work, because sut is always an instance of `Result<int>` or `Result<string>` etc. and the test will fail:

```
  Expected: assignable from <A.Models.Result`1[T]>
  But was:  <A.Models.Result`1[System.Int32]>
```
